File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-4.py
class Solution:
    def findMin(self, nums: List[int]) -> int:
        l, r = 0, len(nums) - 1
        m = 0
        if nums[l] < nums[r]:
            return nums[l]
        while l < r:
            m = (l+r) // 2
            print(m)
            if nums[m] > nums[l]:
                l = m
                m = (l+r) // 2
            elif nums[m] < nums[r]:
                r = m
                m = (l+r) // 2
            else:
                if l == r:
                    break
                l += 1
                m = (l+r) // 2
        print(m)
        return nums[m]

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-4.java
class Solution {
    public int longestConsecutive(int[] nums) {
        if (nums.length == 0) {
            return 0;
        }
        int res = 0;
        Set<Integer> set = new HashSet<>();
        for (int n : nums) {
            set.add(n);
        }

        for (int n : nums) {
            if (!set.contains(n-1)) {
                int val = n-1;
                int cnt = 0;
                while (set.contains(val+1)) {
                    cnt += 1;
                    val += 1;
                }
                res = Math.max(res, cnt);
            }
        }
        return res;
    }
}
